apps/dcos_overlay/src/dcos_overlay_configure.erl

```diff
@@ -12,57 +12,122 @@
 
 -spec(start_link(config()) -> pid()).
 start_link(Config) ->
-   Pid = self(),
-   proc_lib:spawn_link(?MODULE, maybe_configure, [Pid, Config]).
+    Pid = self(),
+    proc_lib:spawn_link(?MODULE, apply_config, [Pid, Config]).
 
 -spec(stop(pid()) -> ok).
 stop(Pid) ->
     unlink(Pid),
     exit(Pid, kill),
     ok.
 
--spec(maybe_configure(pid(), config()) -> term()).
-maybe_configure(Pid, Config) ->
-    {ok, Netlink} = dcos_overlay_netlink:start_link(),
-    try
-        lists:foreach(fun (Overlay) ->
-            maybe_configure_overlay(Netlink, Config, Overlay)
-        end, dcos_overlay_poller:overlays()),
-        Pid ! {dcos_overlay_configure, applied_config, Config}
-    after
-        dcos_overlay_netlink:stop(Netlink)
+-spec(apply_config(pid(), config()) -> ok | {error, term()}).
+apply_config(Pid, Config) ->
+    case apply_config(Config) of
+        ok ->
+            Pid ! {dcos_overlay_configure, applied_config, Config},
+            ok;
+        {error, Error} ->
+            Pid ! {dcos_overlay_configure, failed, Error, Config},
+            {error, Error}
+    end.
+
+-spec(apply_config(config()) -> ok | {error, term()}).
+apply_config(Config) ->
+    case dcos_overlay_netlink:start_link() of
+        {ok, Netlink} ->
+            try maybe_configure(Netlink, Config) of
+                ok ->
+                    ok;
+                {error, Error} ->
+                    lager:error("Failed to apply overlay config ~p due to ~p",
+                        [Config, Error]),
+                    {error, Error}
+            catch
+                Class:Error:StackTrace ->
+                    lager:error("Failed to apply overlay config ~p due to ~p",
+                        [Config, {Class, Error, StackTrace}]),
+                    {error, {Class, Error}}
+            after
+                dcos_overlay_netlink:stop(Netlink)
+            end;
+        {error, Error} ->
+            lager:error("Failed to apply overlay config ~p due to ~p",
+                [Config, Error]),
+            {error, Error}
     end.
 
--spec(maybe_configure_overlay(pid(), config(), jiffy:json_term()) -> ok).
+-spec(maybe_configure(pid(), config()) -> ok | {error, term()}).
+maybe_configure(Netlink, Config) ->
+    lists:foldl(
+        fun(Overlay, ok) ->
+            case maybe_configure_overlay(Netlink, Config, Overlay) of
+                ok ->
+                    ok;
+                {error, Error} ->
+                    lager:error("Failed to configure overlay ~p due to ~p",
+                        [Overlay, Error]),
+                    {error, Error}
+            end;
+           (_, {error, Error}) ->
```

Comment:
Hmm. I see 4- and 8-space indentation here. I guess it depends on from what angle you look at it. :)

Reply:
Your indentation is good, but you can do something like this only with emacs. Otherwise, you have to do 1-, 2-, 3-, 4-space indentation manually. We can't force everyone to use emacs, it's inhumane.

Reply:
I meant, that you use `(` as a reference in the example you provided. Whereas I used `fun` as a reference point, which is I believe consistent with indentation in case of named functions as opposed to anonymous functions in this particular case. But whatever, already adjusted. Consistency across the board is more important.